packages/blade/src/components/Icons/Icons.stories.tsx

```diff
@@ -114,36 +115,50 @@ Icon.args = {
   icon: 'CreditCardIcon',
 };
 
+const IconList = ({
+  IconComponent,
+  iconName,
+  ...args
+}: {
+  IconComponent: IconComponent;
+  iconName: keyof typeof iconMap;
```

Comment:
@anuraghazra can I get the keynames as union type here from the `iconMap` object?

Reply:
Yes, you need to make the iconMap to use as const. 

```
const iconMap = {
...
...
} as const;
```

I will recommend exporting a type from here also
```
type IconNames = keyof typeof iconMap;
```

> because i can see an error in BaseLink.stories where you need to now do `iconMap[(icon as unknown) as IconNames]` instead. (just check that story you'll see the error)

Reply:
did that as const but it still shows `string` instead of union of keys
<img width="407" alt="Screenshot 2023-07-28 at 4 25 28 PM" src="https://github.com/razorpay/blade/assets/11384858/6e57933d-7566-4573-8f6a-8f5bc662f6fa">

Reply:
Did you removed the `const iconMap: Record<string, IconComponent>` type? it needs to be removed so that TS doesn't widen the object type.